src/generic/listctrl.cpp

```diff
@@ -2197,6 +2236,43 @@ void wxListMainWindow::HighlightAll( bool on )
     }
 }
 
+bool wxListMainWindow::HighlightOnly( size_t line, size_t oldLine )
+{
+    const unsigned selCount = GetSelectedItemCount();
+
+    if ( selCount == 1 && IsHighlighted(line) )
+    {
+        return true; // Nothing changed.
+    }
+
+    if ( !IsSingleSel() && selCount != 0 )
```

Comment:
I have some trouble following the logic here. The `else` matching this `if` corresponds to `IsSingleSel() || !selCount` condition and I'm not sure why do test for this. I think things could be a bit more clear if we wrote it like this:
```cpp
if ( IsSingleSel() )
{
    if ( oldLine != (size_t)-1 )
         ...
}
else // multiple-selection
{
    if ( selCount != 0 )
        ...
}
```
wouldn't they?